content/guides/accessibility/color-considerations.mdx

```diff
@@ -0,0 +1,332 @@
+
+Beyond the scope of normative WCAG guidance, a further consideration is if making a decorative icon color contrast compliant is useful for aiding in facilitating understanding, even if it is not required to be.
+
+#### Button border contrast
+
+The contrast of a button's border color compared to the input's background color **does not** need to meet a contrast ratio. Only the [button's text](#button-text-contrast) or [functional icon](#functional-icons) does.
+
+#### Button disabled state contrast
+
+Broadly-speaking, buttons **should not** be disabled until form validation criteria has been met. An example of behaviro to avoid is disabling a submit button until a user has filled out all required fields.
+
+If a button needs to be placed in a disabled state, [it **does not** need to be color contrast compliant](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#inactive-controls). Other WCAG criteria may affect this treatment, however.
+
+Beyond the scope of normative WCAG guidance, a further consideration is there are other ways to visually communicate a disabled button, or to design the system in system in such a way that a button cannot circumstantially be placed in a disabled state. 
+
+While a button's disabled state is exempt from normative compliance, consider that the control may be invisible or unpredictably disappear from the perspective of a user experiencing low vision conditions.
+
+## Patterns that might be confusing
+
+Following are examples of unique GitHub UI that are worth specifically documenting:
+
+### GitHub's search
+
+GitHub's search **visually** looks like an input, but is coded as a button. 
+
+<img src={SearchCollapsed} alt="GitHub's search in its collapsed state. It visually looks like a text input with a magnifying glass icon and placeholder text that reads, 'Type forward slash to search.' An accompanying faux submit button is also present, that looks like a command line prompt awaiting input. A label is pointing to the entire width of the search input and indicates that it is a HTML button element. Cropped screenshot." />
+
+This is because it triggers the dialog that displays the full search interface—a user **cannot** directly enter text into a faux-input field. Because of this, it is **treated like a button** in terms of normative color contrast evaluation.
+
+When the search dialog is displayed, the input that a user types in is treated as a text input for color contrast purposes.
+
+<img src={SearchExpanded} alt="GitHub's search in its expanded state. It visually looks like a text input with a magnifying glass icon. Below it are lists of popular owners and repositories, with the prompt to navigate to each entry. A label is pointing to the text input portion of the search input and indicates that it is a HTML input element with an attribute of 'text.' Cropped screenshot." />
```

Comment:
I don't know if I would include this one because this pattern has a ton of problems and I've recommended several times that this not be a faux input button because of the confusion it has, and the improper aria label that it implies. That's just me on my soapbox though.

Reply:
This was included on request from @lukasoppermann when we last chatted about the document. Not sure if it would do more harm than good to include, but the idea was to pre-empt common questions with something we could link to.

I do see what you mean about not encouraging this approach, however. Do you think there is language we could include to communicate that?

Reply:
Resolving via a pairing session with @ichelsea